enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -230,6 +231,10 @@ public void obligation( long txId ) throws IOException
             {
                 obligationFulfiller.fulfill( txId );
             }
+            catch ( IllegalStateException e )
+            {
+                throw new ComException( "Failed to pull updates", e );
+            }
             catch ( InterruptedException e )
             {
                 throw new IOException( e );
```

Comment:
I still think restoring of interrupted flag might be useful here. When we ask Jetty to stop it tries to stop all threads in it's pool using interrupts. So `InterruptedException` might be thrown from `obligationFulfiller.fulfill( txId )`. `IOException` will just kill the running transaction but thread would not know that it is time to stop and might try to perform some work. Any thoughts?